service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/bqdataset/ControlledBigQueryDatasetResource.java

```diff
@@ -236,15 +235,8 @@ public ControlledBigQueryDatasetResource.Builder common(ControlledResourceFields
     }
 
     public ControlledBigQueryDatasetResource.Builder datasetName(String datasetName) {
-      try {
-        // If the user doesn't specify a dataset name, we will use the resource name by default.
-        // But if the resource name is not a valid dataset name, we will need to generate a unique
-        // dataset id.
-        ResourceValidationUtils.validateBqDatasetName(datasetName);
-        this.datasetName = datasetName;
-      } catch (InvalidReferenceException e) {
-        this.datasetName = generateUniqueDatasetId();
-      }
+      this.datasetName =
+          ControlledBigQueryDatasetHandler.getHandler().generateCloudName(null, datasetName);
```

Comment:
can you move this to the call site and here we just do this.datasetName = datasetName;

also only call generate when it's null